cdap-security/src/main/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcer.java

```diff
@@ -60,6 +63,12 @@ public void enforce(EntityId entity, Principal principal, Action action) throws
   }
 
   private void doEnforce(EntityId entity, Principal principal, Set<Action> actions) throws Exception {
+    if (entity instanceof NamespacedEntityId &&
```

Comment:
What if the principal is 'ali' and the UserGroupInformation is also 'ali'? Then, 'ali' will be allowed to access entities in the system namespace, such as system dataset instances?